x-pack/test_utils/README.md

```diff
@@ -0,0 +1,255 @@
+"extensions" window in Chrome then drag and drop the `test_subjects_finder.crx` file on the window.
+
+You can specify a DOM node (the tree "root") from which the test subjects will be found. If you don't specify any, the document `<body>` 
+will be used. The output format can either be `Typescript` (to export a string union type) or `List`. If you use Typescript, 
+the union type will give you autocomplete on the test subjects in your test, you just need to provide the union type when registring the Testbed.
+
+```ts
+// my_component.helpers.ts
+
+type TestSubjects = 'addButton'
+ | 'autoFollowPatternForm';
+
+export const setup = registerTestBed<TestSubjects>(MyComponent);
+```
+
+## API
+
+## `registerTestBed(Component [, testBedConfig])`
+
+Instantiate a new TestBed to test a React component. The arguments it receives are
+
+- `Component` (the React component to test)
+- `testBedConfig` (optional). An optional Testbed configuration object.
+
+**@returns** A function to instantiate and mount the component.
+
+### `testBedConfig`
+
+The `testBedConfig` has the following properties (all **optional**)
+
+- `defaultProps` The default props to pass to the mounted component. Those props can be overriden when calling the `setup([props])` callback
+- `memoryRouter` Configuration object for the react-router `MemoryRouter` with the following properties
+  - `wrapComponent` Flag to provide or not a `MemoryRouter`. If set to `false`, there won't be any router and the component won't be added on a `<Route />`. (default: `true`)
+  - `initialEntries` The React Router **initial entries** setting. (default: `['/']`. [see doc](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/MemoryRouter.md))
+  - `initialIndex` The React Router **initial index** setting (default: `0`)
+  - `componentRoutePath` The route **path** for the mounted component (default: `"/"`)
+  - `onRouter` A callBack that will be called with the React Router instance when the component is mounted
+- `store` An redux store. You can also provide a function that returns a store. (default: `null`)
+
+
+## `setup([props])`
+
+When registering a Testbed, you receive in return a setup function. This function accepts one optional argument:
+
+- `props` (optional) Props to pass to the mounted component.
+
+```js
+describe('<RemoteClusterList />', () => {
+  test('it should be green', () => {
+    // Will mount <RemoteClusterList color="green" />
+    const { component } = setup({ color: 'green' });
+    ...
+  });
+});
+```
+
+**@returns** An object with the following **properties** and **helpers** for the testing
+
+#### `component`
+
+The mounted component. It is an enzyme **reactWrapper**.
+
+#### `exists(testSubject)`
+
+Pass it a `data-test-subj` and it will return true if it exists or false if it does not exist. You can provide a nested path to access the
```

Comment:
do you think it would be helpful to link back to this doc, which discusses `testSubjects`? https://github.com/elastic/kibana/blob/master/docs/development/core/development-functional-tests.asciidoc

Reply:
Thanks for raising this, it's a good point. Instead of pointing to the documentation of the functional tests, I decided to include the part of the test subjects in this README. What do you think?

Reply:
Yeah, looks good. Thanks for adding that!